tests/FunctionCallTest.php

```diff
@@ -1931,7 +1931,12 @@ function a($b): int
                       return $b["a"];
                     }
 
-                    a(["a" => "hello"]);',
+                    function b(string $str): string
+                    {
+                        return $str;
+                    }
+
+                    a(["a" => b("hello")]);',
```

Comment:
Non-int literal strings are no longer coercible to int, I think this is an improvement. Without this change it now reports `InvalidArgument`.

Reply:
This would be a big BC break and I'm not sure I see the point. Maybe the term coercion is badly chosen in this case. The actual issue doesn't mention coercion: https://psalm.dev/docs/running_psalm/issues/InvalidScalarArgument/

Reply:
`InvalidScalarArgument` originally came about because, in non-strict mode, PHP is more lenient with those scalar type conversions.

If you take the example given in the issue link above, and add a `strict_types` declare you get `InvalidArgument` ([example](https://psalm.dev/r/83cfce18a5)). But that's not universal — docblock types are [treated differently than runtime typehints](https://psalm.dev/r/001ac14f96). That lines up to `InvalidScalarArgument`'s original intent — to warn about likely runtime errors — but it can make some bugs seem less harmful even though they still lead to runtime errors (there's just a level of indirection).

So I think there might be a solid argument for reconsidering `InvalidScalarArgument` (as part of a major version change).

I'll draft a PR with that change for Psalm 5 to see what people think. Might even be worth a Twitter poll to take people's temperature.

Reply:
`"1"` is coercible to `int` and would report `InvalidScalarArgument` in this context, the way I did the type comparisons literals like `"hello"` are no longer coercible to `int`. I suppose either way `InvalidScalarArgument` still applies as it is described, but given that the string is definitely not an integer-string I would think it would make more sense to emit `InvalidArgument`.